src/meta/src/barrier/context/context_impl.rs

```diff
@@ -196,12 +197,20 @@ impl CommandContext {
 
             Command::DropStreamingJobs {
                 unregistered_state_table_ids,
+                dropped_sink_in_existing_table,
                 ..
             } => {
                 barrier_manager_context
                     .hummock_manager
                     .unregister_table_ids(unregistered_state_table_ids.iter().cloned())
                     .await?;
+                if !dropped_sink_in_existing_table.is_empty() {
+                    barrier_manager_context
+                        .metadata_manager
+                        .catalog_controller
+                        .update_table_remove_incoming_sinks(dropped_sink_in_existing_table)
```

Comment:
For ddl to drop a streaming job, we will remove it from catalog before we yield the barrier, to ensure that we don't have to wait for barrier collection to drop any problematic jobs. We should change to update the `incoming_sinks` in the same transaction to remove the sinks.

Besides, I'm not sure whether the `incoming_sinks` field should be deprecated, because we already store a `target_table` field in `sink`, and we can always restore the `incoming_sinks` of a table from the sinks' `target_table`. In this way, we don't have to maintain the consistency between the two fields.

Reply:
+1 for updating all catalogs in the same transaction before issuing any barrier. Another benefit is that no additional cleaning work is required during the recovery process.